nifi-toolkit/nifi-toolkit-kafka-migrator/src/main/java/org/apache/nifi/toolkit/kafkamigrator/service/KafkaMigrationService.java

```diff
@@ -0,0 +1,110 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.toolkit.kafkamigrator.service;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.nifi.toolkit.kafkamigrator.migrator.Migrator;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+
+import javax.xml.xpath.XPath;
+import javax.xml.xpath.XPathConstants;
+import javax.xml.xpath.XPathExpressionException;
+import javax.xml.xpath.XPathFactory;
+import java.util.Map;
+
+public class KafkaMigrationService {
+
+    private final static String REGEX_FOR_REPLACEABLE_PROCESSOR_NAMES = "(Get|Put|Consume|Publish)Kafka(Record)?(_0_1\\d)?";
+    private final static String NEW_KAFKA_PROCESSOR_VERSION = "_2_0";
+    private final static String ARTIFACT = "nifi-kafka-2-0-nar";
+    private final static String PATH_FOR_ARTIFACT = "bundle/artifact";
+    protected static final boolean IS_VERSION_EIGHT_PROCESSOR = Boolean.TRUE;
+    protected static final boolean IS_NOT_VERSION_EIGHT_PROCESSOR = Boolean.FALSE;
+
+    protected String path;
+    protected String pathForClass;
+
+    public void replaceKafkaProcessors(final Document document, final Map<String, String> arguments) throws XPathExpressionException {
+        final KafkaFlowMigrationService flowMigrationService = new KafkaFlowMigrationService();
+        final KafkaTemplateMigrationService templateMigrationService = new KafkaTemplateMigrationService();
+
+        flowMigrationService.replaceKafkaProcessors(document, arguments);
+        templateMigrationService.replaceKafkaProcessors(document, arguments);
+    }
+
+    protected void replaceProcessors(Document document, Map<String, String> arguments) throws XPathExpressionException {
+        Migrator migrator;
+        final XPath xPath = XPathFactory.newInstance().newXPath();
+
+        final NodeList processors = (NodeList) xPath.evaluate(path, document, XPathConstants.NODESET);
+        for (int i = 0; i < processors.getLength(); i++) {
+            final Node processor = processors.item(i);
+            final String className = xPath.evaluate(pathForClass, processor);
+            final String processorName = StringUtils.substringAfterLast(className, ".");
+
+            if (replaceableKafkaProcessor(processorName)) {
+                if (processorName.contains("Publish")) {
+                    migrator = createPublishMigrator(arguments);
+                } else if (processorName.contains("Put")) {
+                    migrator = createVersionEightPublishMigrator(arguments);
+                } else if (processorName.contains("Consume")) {
+                    migrator = createConsumeMigrator(arguments);
+                } else {
+                    migrator = createVersionEightConsumeMigrator(arguments);
+                }
+
+                migrator.configureProperties(processor);
```

Comment:
I would consider introduce only one method (migrateProcessor(processor, arguments) ?) for the migrator interface and call only that here. In the child this way maybe we could avoid empty methods and call exactly what is needed?